### ISSUE TITLE
boost-fiber:x64-windows-static build fails

### ISSUE THREAD

**Issue by dimiboi** [microsoft/vcpkg#6809]
```
Building package boost-fiber[core]:x64-windows-static...
-- Downloading https://github.com/boostorg/fiber/archive/boost-1.70.0.tar.gz...
-- Extracting source C:/vcpkg/downloads/boostorg-fiber-boost-1.70.0.tar.gz
-- Using source at C:/vcpkg/buildtrees/boost-fiber/src/ost-1.70.0-57b4c46dd0
-- Building x64-windows-static-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:56 (message):
    Command failed: C:/vcpkg/installed/x64-windows-static/tools/boost-build/b2.exe;--stagedir=C:/vcpkg/buildtrees/boost-fiber/x64-windows-static-rel/stage;--build-dir=C:/vcpkg/buildtrees/boost-fiber/x64-windows-static-rel;--user-config=C:/vcpkg/buildtrees/boost-fiber/x64-windows-static-rel/user-config.jam;-j2;--debug-configuration;--debug-building;--debug-generators;--disable-icu;--ignore-site-config;--hash;-q;-sZLIB_INCLUDE="C:/vcpkg/installed/x64-windows-static/include";-sBZIP2_INCLUDE="C:/vcpkg/installed/x64-windows-static/include";threading=multi;threadapi=win32;runtime-link=static;link=static;address-model=64;architecture=x86;toolset=msvc;-sZLIB_BINARY=zlib;-sZLIB_LIBPATH="C:/vcpkg/installed/x64-windows-static/lib";-sBZIP2_BINARY=bz2;-sBZIP2_LIBPATH="C:/vcpkg/installed/x64-windows-static/lib";variant=release;debug-symbols=on
    Working Directory: C:/vcpkg/buildtrees/boost-fiber/src/ost-1.70.0-57b4c46dd0/build
    Error code: 1
    See logs for more information:
      C:\vcpkg\buildtrees\boost-fiber\build-x64-windows-static-rel-out.log

Call Stack (most recent call first):
  installed/x64-windows-static/share/boost-build/boost-modular-build.cmake:278 (vcpkg_execute_required_process)
  ports/boost-fiber/portfile.cmake:19 (boost_modular_build)
  scripts/ports.cmake:71 (include)


Error: Building package boost-fiber:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: boost-fiber:x64-windows-static
  Vcpkg version: 2018.11.23-nohash

Additionally, attach any relevant sections from the log files above.
```

**Comment by wangli28** [comment]
Hi @dimiboi, thanks for posting this issue. Can you provide your environment information and Visual Studio version?

**Comment by dimiboi** [comment]
Hi, @wangli28, I am using [Azure Pipelines VS2017 image](https://github.com/microsoft/azure-pipelines-image-generation/blob/master/images/win/Vs2017-Server2016-Readme.md). I get the same error with pre-installed Vcpkg and when I install the latest Vcpkg.

**Comment by FerzanVivec** [comment]
Fiber also fails to build on my system.

```
C:\ProgrammingLibraries\vcpkg>vcpkg install boost-fiber
The following packages will be built and installed:
    boost-fiber[core]:x64-windows
Starting package 1/1: boost-fiber:x64-windows
Building package boost-fiber[core]:x64-windows...
-- Using cached C:/ProgrammingLibraries/vcpkg/downloads/boostorg-fiber-boost-1.70.0.tar.gz
-- Using source at C:/ProgrammingLibraries/vcpkg/buildtrees/boost-fiber/src/ost-1.70.0-57b4c46dd0
-- Building x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:58 (message):
    Command failed: C:/ProgrammingLibraries/vcpkg/installed/x64-windows/tools/boost-build/b2.exe --stagedir=C:/Programmi
ngLibraries/vcpkg/buildtrees/boost-fiber/x64-windows-rel/stage --build-dir=C:/ProgrammingLibraries/vcpkg/buildtrees/boos
t-fiber/x64-windows-rel --user-config=C:/ProgrammingLibraries/vcpkg/buildtrees/boost-fiber/x64-windows-rel/user-config.j
am -j8 --debug-configuration --debug-building --debug-generators --disable-icu --ignore-site-config --hash -q -sZLIB_INC
LUDE="C:/ProgrammingLibraries/vcpkg/installed/x64-windows/include" -sBZIP2_INCLUDE="C:/ProgrammingLibraries/vcpkg/instal
led/x64-windows/include" threading=multi threadapi=win32 runtime-link=static link=static address-model=64 architecture=x
86 toolset=msvc -sZLIB_BINARY=zlib -sZLIB_LIBPATH="C:/ProgrammingLibraries/vcpkg/installed/x64-windows/lib" -sBZIP2_BINA
RY=bz2 -sBZIP2_LIBPATH="C:/ProgrammingLibraries/vcpkg/installed/x64-windows/lib" variant=release debug-symbols=on
    Working Directory: C:/ProgrammingLibraries/vcpkg/buildtrees/boost-fiber/src/ost-1.70.0-57b4c46dd0/build
    Error code: 1
    See logs for more information:
      C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\build-x64-windows-rel-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/boost-build/boost-modular-build.cmake:278 (vcpkg_execute_required_process)
  ports/boost-fiber/portfile.cmake:19 (boost_modular_build)
  scripts/ports.cmake:73 (include)


Error: Building package boost-fiber:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: boost-fiber:x64-windows
  Vcpkg version: 2019.06.21-nohash

Additionally, attach any relevant sections from the log files above.
```
Here's the log file:
```
notice: found boost-build.jam at C:/ProgrammingLibraries/vcpkg/installed/x64-windows/tools/boost-build/boost-build.jam
notice: loading Boost.Build from C:/ProgrammingLibraries/vcpkg/installed/x64-windows/tools/boost-build/src/kernel
notice: Site configuration files will be ignored due to the
notice: --ignore-site-config command-line option.
notice: Loading explicitly specified user configuration file:
    C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\x64-windows-rel\user-config.jam
notice: Searching 'C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\x64-windows-rel' for user-config configuration file 'user-config.jam'.
notice: Loading user-config configuration file 'user-config.jam' from 'C:/ProgrammingLibraries/vcpkg/buildtrees/boost-fiber/x64-windows-rel'.
notice: [msvc-cfg] msvc-14.0 detected, command: 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\cl.exe'
notice: will use 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\cl.exe' 'cl.exe' for msvc, condition <toolset>msvc-14.0
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>/<address-model>', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>/<address-model>', setup: 'x86'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>/<address-model>32', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>/<address-model>32', setup: 'x86'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>x86/<address-model>', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>x86/<address-model>', setup: 'x86'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>x86/<address-model>32', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>x86/<address-model>32', setup: 'x86'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>/<address-model>64', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>/<address-model>64', setup: 'amd64'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>x86/<address-model>64', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>x86/<address-model>64', setup: 'amd64'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>ia64/<address-model>', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>ia64/<address-model>', setup: 'x86_ia64'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>ia64/<address-model>64', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>ia64/<address-model>64', setup: 'x86_ia64'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>arm/<address-model>', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>arm/<address-model>', setup: 'x86_arm'
notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>arm/<address-model>32', setup: 'C:/ProgrammingLibraries/vcpkg/installed/x64-windows/share/boost-build/nothing.bat' notice: [msvc-cfg] condition: '<toolset>msvc-14.0/<architecture>arm/<address-model>32', setup: 'x86_arm'
building project Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>  ('object(project-target)@290') with <address-model>64 <architecture>x86 <debug-symbols>on <link>static <runtime-link>static <threadapi>win32 <threading>multi <toolset>msvc <variant>release

     Building target '../build/boost_fiber'
         Build request: object(property-set)@404 <address-model>64 <architecture>x86 <debug-symbols>on <link>static <runtime-link>static <threadapi>win32 <threading>multi <toolset>msvc <variant>release
         Command line free features:
         Target requirements: <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@323.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@324.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@325.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@326.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@327.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@328.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@329.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@330.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@331.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@332.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@333.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@334.check <conditional>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0\build>%object(check-target-builds-worker)@335.check <define>BOOST_ALL_NO_LIB=1 <define>BOOST_FIBERS_SOURCE <include>../include&&/C:/ProgrammingLibraries/vcpkg/installed/x64-windows/include <library>/boost/context//boost_context <library>/boost/filesystem//boost_filesystem <link>shared:<define>BOOST_FIBERS_DYN_LINK=1 <link>shared:<library>/boost/context//boost_context <relevant>build:<relevant>address-model <relevant>build:<relevant>architecture <relevant>build:<relevant>cxxstd <relevant>build:<relevant>cxxstd-dialect <relevant>build:<relevant>target-os <relevant>build:<relevant>toolset <relevant>build:<relevant>toolset-msvc:vendor <relevant>build:<relevant>toolset-msvc:version <relevant>cxxflags:<relevant>segmented-stacks <relevant>cxxflags:<relevant>toolset <relevant>define:<relevant>link <relevant>define:<relevant>target-os <relevant>library:<relevant>link <relevant>linkflags:<relevant>target-os <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <tag>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>%Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>.tag <target-os>solaris:<linkflags>-llgrp <target-os>windows:<define>_WIN32_WINNT=0x0601 <threading>multi <toolset>clang,<segmented-stacks>on:<cxxflags>-DBOOST_USE_SEGMENTED_STACKS <toolset>clang,<segmented-stacks>on:<cxxflags>-fsplit-stack <toolset>gcc,<segmented-stacks>on:<cxxflags>-DBOOST_USE_SEGMENTED_STACKS <toolset>gcc,<segmented-stacks>on:<cxxflags>-fsplit-stack <toolset>msvc,<runtime-link>shared:<threading>multi

         Building target '/C:/ProgrammingLibraries/vcpkg/buildtrees/boost-fiber/src/ost-1.70.0-57b4c46dd0/build/config/checks/cxx11_auto_declarations'
             Build request: object(property-set)@412 <address-model>64 <architecture>x86 <target-os>windows <toolset-msvc:version>14.0 <toolset>msvc
             Command line free features:
             Target requirements: <define>BOOST_ALL_NO_LIB=1 <define>BOOST_FIBERS_SOURCE <define>TEST_BOOST_NO_CXX11_AUTO_DECLARATIONS <include>../include&&/C:/ProgrammingLibraries/vcpkg/installed/x64-windows/include <library>/boost/context//boost_context <library>/boost/filesystem//boost_filesystem <link>shared:<define>BOOST_FIBERS_DYN_LINK=1 <relevant>cxxflags:<relevant>segmented-stacks <relevant>cxxflags:<relevant>toolset <relevant>define:<relevant>link <relevant>define:<relevant>target-os <relevant>linkflags:<relevant>target-os <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <tag>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>%Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>.tag <target-os>solaris:<linkflags>-llgrp <target-os>windows:<define>_WIN32_WINNT=0x0601 <threading>multi <toolset>clang,<segmented-stacks>on:<cxxflags>-DBOOST_USE_SEGMENTED_STACKS <toolset>clang,<segmented-stacks>on:<cxxflags>-fsplit-stack <toolset>gcc,<segmented-stacks>on:<cxxflags>-DBOOST_USE_SEGMENTED_STACKS <toolset>gcc,<segmented-stacks>on:<cxxflags>-fsplit-stack <toolset>msvc,<runtime-link>shared:<threading>multi

             Common properties:  <address-model>64 <architecture>x86 <asynch-exceptions>off <debug-store>object <debug-symbols>on <define>BOOST_ALL_NO_LIB=1 <define>BOOST_FIBERS_DYN_LINK=1 <define>BOOST_FIBERS_SOURCE <define>TEST_BOOST_NO_CXX11_AUTO_DECLARATIONS <define>_WIN32_WINNT=0x0601 <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <hardcode-dll-paths>true <host-os>windows <include>../include&&/C:/ProgrammingLibraries/vcpkg/installed/x64-windows/include <inlining>off <install-dependencies>off <library>/boost/context//boost_context <library>/boost/filesystem//boost_filesystem <link>shared <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>cxxflags:<relevant>segmented-stacks <relevant>cxxflags:<relevant>toolset <relevant>define:<relevant>link <relevant>define:<relevant>target-os <relevant>linkflags:<relevant>target-os <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <rtti>on <runtime-debugging>on <runtime-link>shared <stdlib>native <strip>off <suppress-import-lib>false <symlink-location>project-relative <tag>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>%Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>.tag <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>14.0 <toolset>msvc <user-interface>console <variant>debug <vectorize>off <warnings-as-errors>off <warnings>on <windows-api>desktop

             Building target '../boost_context'
                 Build request: object(property-set)@2155 <address-model>64 <architecture>x86 <asynch-exceptions>off <debug-store>object <debug-symbols>on <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <inlining>off <link>shared <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>variant <rtti>on <runtime-debugging>on <runtime-link>shared <stdlib>native <strip>off <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>14.0 <toolset>msvc <variant>debug <vectorize>off <warnings-as-errors>off <warnings>on <windows-api>desktop
                 Command line free features:
                 Target requirements: <define>BOOST_ALL_NO_LIB=1 <file>/boost//C:/ProgrammingLibraries/vcpkg/installed/x64-windows/debug/lib/boost_context-vc140-mt-gd.lib <include>../include&&/C:/ProgrammingLibraries/vcpkg/installed/x64-windows/include <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <tag>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>%Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>.tag <toolset>msvc,<runtime-link>shared:<threading>multi <variant>debug

                 Common properties:  <address-model>64 <architecture>x86 <asynch-exceptions>off <debug-store>object <debug-symbols>on <define>BOOST_ALL_NO_LIB=1 <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <file>/boost//C:/ProgrammingLibraries/vcpkg/installed/x64-windows/debug/lib/boost_context-vc140-mt-gd.lib <hardcode-dll-paths>true <host-os>windows <include>../include&&/C:/ProgrammingLibraries/vcpkg/installed/x64-windows/include <inlining>off <install-dependencies>off <link>shared <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <relevant>variant <rtti>on <runtime-debugging>on <runtime-link>shared <stdlib>native <strip>off <suppress-import-lib>false <symlink-location>project-relative <tag>@Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>%Jamfile<C:\ProgrammingLibraries\vcpkg\buildtrees\boost-fiber\src\ost-1.70.0-57b4c46dd0>.tag <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>14.0 <toolset>msvc <user-interface>console <variant>debug <vectorize>off <warnings-as-errors>off <warnings>on <windows-api>desktop
error: at C:/ProgrammingLibraries/vcpkg/installed/x64-windows/tools/boost-build/src/kernel\modules.jam:107
error: Unable to find file or target named
error:     '/boost//C:/ProgrammingLibraries/vcpkg/installed/x64-windows/debug/lib/boost_context-vc140-mt-gd.lib'
error: referred to from project at
error:     '..'
```




Cmake version:

```
C:\ProgrammingLibraries\vcpkg>cmake --version
cmake version 3.15.0-rc3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
Visual Studio version info:

```
Microsoft Visual Studio Enterprise 2015
Version 14.0.25420.01 Update 3
Microsoft .NET Framework
Version 4.7.03062

Installed Version: Enterprise
```

Here's the Triplet info:

```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE static)
set(VCPKG_LIBRARY_LINKAGE static)
set(VCPKG_BUILD_TYPE release)
```

**Comment by josch** [comment]
It seems I ran into a similar issue in #7559.

@dimiboi what is the content of `C:\vcpkg\buildtrees\boost-fiber\build-x64-windows-static-rel-out.log` after your build failed? Did you only try to perform a static build or did you happen to change the `VCPKG_BUILD_TYPE` to `release` just as @FerzanVivec did?

@FerzanVivec does your problem persist if you do not set `VCPKG_BUILD_TYPE` to `release` and thus build debug *and* release builds? In #7559 I see a very similar issue when setting `VCPKG_BUILD_TYPE` but I'm not doing static builds.

**Comment by dimiboi** [comment]
The latest vcpkg version works in VS2017 pipeline.

**Comment by josch** [comment]
@dimiboi Okay, but did you or did you not also touch the `VCPKG_BUILD_TYPE` in your triplet?

I just set up a fresh Windows 10 1903 VM, installed visual studio community 2019, cloned vcpkg HEAD and am still unable to compile boost-fiber if I set `VCPKG_BUILD_TYPE` to `release`. The only reason I'm mentioning this maybe orthogonal issue here is, that my error output looks just the same as the one posted here.

**Comment by dimiboi** [comment]
@josch, I did not touch `VCPKG_BUILD_TYPE`.

**Comment by whozcraig** [comment]
Fyi, i had a similar issue crop up with the adoption of boost 1.70 in the vcpkg collective, #7295 , that has yet to see any resolution or activity once marked with needs-repo. I provided logs, but I'm curious. The error in your log is a failure to find this file during dependency linking:

> C:/ProgrammingLibraries/vcpkg/installed/x64-windows/debug/lib/boost_context-vc140-mt-gd.lib

If you navigate to that folder, is there a lib there, but the actual file name is tagged with vc6 rather than vc140? I ask because that's the symptoms I'm seeing on my environment. Just curious. Alas I don't know enough about how triplets work, or how to circumvent this issue, so I'll be of little help there. But at least misery loves company.

**Comment by josch** [comment]
@whozcraig thanks! This made me look closer at the error message.

The file `installed/x64-windows/debug/lib/boost_context-vc140-mt-gd.lib` does indeed exist. What is actually wrong can be found in this part of the error message:

```
error: Unable to find file or target named
error:     '/boost//C:/ProgrammingLibraries/vcpkg/installed/x64-windows/debug/lib/boost_context-vc140-mt-gd.lib'
error: referred to from project at
error:     '..'
```

Notice the `/boost//` prefix in front of `C:/...`. Obviously such a path does not exist. I assume some paths are simply wrongly concatenated here.

**Comment by dimiboi** [comment]
@josch, if I add `set(VCPKG_BUILD_TYPE release)` the build fails.

Here's Azure Pipeline I used to reproduce the issue:

```yaml
trigger:
  - '*'

pool:
  vmImage: 'windows-2019'

variables:
  - name: vcpkgDir
    value: vcpkg
  - name: triplet
    value: x64-windows-static

steps:
  - powershell: |
      git clone https://github.com/microsoft/vcpkg.git $(vcpkgDir)
      cd $(vcpkgDir)
      .\bootstrap-vcpkg.bat
      Add-Content -Path ".\triplets\$(triplet).cmake" "set(VCPKG_BUILD_TYPE release)"
      .\vcpkg install boost:$(triplet)
    displayName: 'Install Boost via Vcpkg'
```

**Comment by josch** [comment]
@dimiboi does the same happen if you choose `x64-windows` (without static)? Is the error message you get the same as the one I posted above in each case?

I'm asking because I'm still not sure whether this issue here is the same as the one in #7559 or not.

**Comment by dimiboi** [comment]
@josch, same thing in non-static build:

```
Building package liblzma[core]:x64-windows...
-- Downloading https://github.com/xz-mirror/xz/archive/v5.2.4.tar.gz...
-- Extracting source D:/a/1/s/vcpkg/downloads/xz-mirror-xz-v5.2.4.tar.gz
-- Applying patch enable-uwp-builds.patch
-- Using source at D:/a/1/s/vcpkg/buildtrees/liblzma/src/v5.2.4-66bfe33102
-- Configuring x64-windows
-- Installing: D:/a/1/s/vcpkg/packages/liblzma_x64-windows/share/liblzma/copyright
-- Performing post-build validation
The folder /include is empty or not present. This indicates the library was not correctly installed.
Found 1 error(s). Please correct the portfile:
    D:\a\1\s\vcpkg\ports\liblzma\portfile.cmake
-- Performing post-build validation done
Error: Building package liblzma:x64-windows failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: liblzma:x64-windows
  Vcpkg version: 2019.07.18-nohash
```

**Comment by josch** [comment]
@dimiboi but this error output is different from what I am seeing and seems to be unrelated to the boost-fiber issue? For me, liblzma compiled just fine.

**Comment by dimiboi** [comment]
@josch, yes, it is indeed different. Perhaps, a separate issue?

**Comment by whozcraig** [comment]
> Notice the `/boost//` prefix in front of `C:/...`. Obviously such a path does not exist. I assume some paths are simply wrongly concatenated here.

yeah, I noticed that. I just checked on all of my dev rigs (There are seven total, two used by me, five more from other developers, and they all have the same problem I stated. 

error: Unable to find file or target named
error:     '/boost//C:/vcpkg/installed/x86-windows/lib/boost_chrono-vc140-mt.lib'
error: referred to from project at
error:     '..'

Belaying the odd path for a moment, and just going to that folder anyway, I find this file:

boost_chrono-vc6-mt.lib

In fact, **all** the built libraries in that folder (which I assume only built because they have no other lib-based dependencies and are strictly source/header based) are similarly named. The **vc6** is the part i question, obviously. 

I cannot fathom for the life of me how this isn't reproducible in-general. Every development Vs2015u3 rig we have at our disposal exhibits the same problem, and it has been there for almost a month now. 

Anyway, best of luck to you. Hopefully MS will dig deep and give vcpkg some added resources. It really is/was slick when it work(ed),  and I sure would hate to have to ditch it for something else.
